api/tests/test_token_checker.py

```diff
@@ -0,0 +1,397 @@
+            'jti': '714892f5-014f-43ad-bea0-fa47579db222'
+        })
+
+        # Add a payload without 'exp' to the payload_list
+        # to test we reject these (as we are choosing to)
+        payload_list.append({
+            'iss': 'https://iam-test.indigo-datacloud.eu/',
+            'jti': '098cb343-c45e-490d-8aa0-ce1873cdc5f8',
+            'iat': int(time.time()) - 2000000,
+            'sub': CLIENT_ID
+        })
+
+        # Add a payload without 'iat'
+        # to test we reject these (as we are choosing to)
+        payload_list.append({
+            'iss': 'https://iam-test.indigo-datacloud.eu/',
+            'jti': '098cb343-c45e-490d-8aa0-ce1873cdc5f8',
+            'sub': CLIENT_ID,
+            'exp': int(time.time()) + 200000
+        })
+
+        # Add a payload with an 'iat' and 'exp' in the past
+        # (e.g. they have expired) to test we are
+        # rejecting these
+        payload_list.append({
+            'iss': 'https://iam-test.indigo-datacloud.eu/',
+            'jti': '098cb343-c45e-490d-8aa0-ce1873cdc5f8',
+            'iat': int(time.time()) - 2000000,
+            'sub': CLIENT_ID,
+            'exp': int(time.time()) - 200000
+        })
+
+        # Add a payload with an 'iat' and 'exp' in the future
+        # to test we are rejecting these (as we should as they
+        # are not yet valid)
+        payload_list.append({
+            'iss': 'https://iam-test.indigo-datacloud.eu/',
+            'jti': '098cb343-c45e-490d-8aa0-ce1873cdc5f8',
+            'iat': int(time.time()) + 200000,
+            'sub': CLIENT_ID,
+            'exp': int(time.time()) + 2000000
+        })
+
+        for payload in payload_list:
+            # Assert the underlying helper method reponsible for
+            # checking temporal validity returns False when passed
+            # temporally invalid payloads
+            self.assertFalse(
+                self._token_checker._is_token_json_temporally_valid(payload),
+                "Payload %s should not be accepted!" % payload
+            )
+
+            # Assert the wrapper method valid_token_to_id returns
+            # None when passed temporally invalid tokens
+            token = self._create_token(payload, PRIVATE_KEY)
+            self.assertEqual(
+                self._token_checker.valid_token_to_id(token), None,
+                "Token with payload %s should not be accepted!" % payload
+            )
+
+    def test_garbage_token(self):
+        """Test a garbage token is rejected."""
+        token = 'ffnnsdifsdjofjfosdjfodsjfosdjofj'
+        result = self._token_checker.valid_token_to_id(token)
+        self.assertEqual(result, None)
+
+    def test_http_issuer_ban(self):
+        """Test a HTTP issuer is rejected."""
+        self.assertEqual(
+            self._token_checker._check_token_not_revoked(None,
+                                                         'http://idc.org'),
+            None
+        )
```

Comment:
This is somewhat clearer. Not much you can do when you have really long method names and really short arguments.

Reply:
indeed :(